src/commonMain/kotlin/baaahs/sm/brain/BrainManager.kt

```diff
@@ -246,7 +250,7 @@ class BrainManager(
         private fun deliverShaderMessage() {
             val message = BrainShaderMessage(pixelBuffer.brainShader, pixelBuffer).toBytes()
             val now = clock.now()
-            if (brainController.lastErrorAt?.isBefore(now - waitPeriodAfterNetworkError) != false) {
```

Comment:
Does UDP `send()` actually return errors for closed ports or unresponsive hosts? I don't think it's blocking; maybe it remembers if it got a UDP fail previously and fails on the next call?

Reply:
From `DatagramSocket.send(DatagramPacket p)`:

```
     * @throws     PortUnreachableException may be thrown if the socket is connected
     *             to a currently unreachable destination. Note, there is no
     *             guarantee that the exception will be thrown.
```

I kinda suspect we won't ever hear about it if the host just disappears…

Reply:
I actually see No Route To Host on my local network if I unplug the brain